packages/keystone/plugins/utils/address-service-client/AddressServiceClient.js

```diff
@@ -0,0 +1,102 @@
+const fetch = require('node-fetch')
+
+/**
+ * @typedef {Object} AddressServiceParams
+ * @property {string?} geo
+ * @property {number?} count Number of results to return (20 by default)
+ * @property {string?} context
+ */
+
+/**
+ *
+ */
+class AddressServiceClient {
+    /**
+     * @param {string} url The address service url (root)
+     * @param {AddressServiceParams?} params Additional parameters. Ability to set query parameters for all queries
+     */
+    constructor (url, params = {}) {
+        if (!url) {
+            throw new Error('The `url` parameter is mandatory')
+        }
+
+        this.url = url
+
+        // The next fields allow to define some query parameters for all queries made by this client instance
+        const { geo = null, count = 20, context = null } = params
+
+        this.possibleUrlParams = {
+            geo,
+            count,
+            context,
+        }
+    }
+
+    /**
+     * @param {AddressServiceParams} params
+     * @returns {string}
+     * @private
+     */
+    urlifyParams (params) {
```

Comment:
Is that same with native `URLSearchParams`?
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams